crypto/cms/cms_env.c

```diff
@@ -909,20 +977,20 @@ BIO *cms_EnvelopedData_init_bio(const CMS_ContentInfo *cms)
 
     /* If error or no cipher end of processing */
 
-    if (!ret || !ec->cipher)
+    if (!ret)
         return ret;
 
     /* Now encrypt content key according to each RecipientInfo type */
 
     rinfos = cms->d.envelopedData->recipientInfos;
 
     for (i = 0; i < sk_CMS_RecipientInfo_num(rinfos); i++) {
-        ri = sk_CMS_RecipientInfo_value(rinfos, i);
-        if (CMS_RecipientInfo_encrypt(cms, ri) <= 0) {
-            CMSerr(CMS_F_CMS_ENVELOPEDDATA_INIT_BIO,
-                   CMS_R_ERROR_SETTING_RECIPIENTINFO);
-            goto err;
-        }
+         ri = sk_CMS_RecipientInfo_value(rinfos, i);
+         if (CMS_RecipientInfo_encrypt(cms, ri) <= 0) {
+             CMSerr(CMS_F_CMS_ENVELOPEDDATA_ENCRYPTION_INIT_BIO,
```

Comment:
just use 0 if its a new one.. instead of CMS_F_CMS_ENVELOPEDDATA_ENCRYPTION_INIT_BIO